src/pages/CustomLessonsDisplay/CustomLessonsDisplay.js

```diff
@@ -1,67 +1,86 @@
 import LessonDateDisplay from "../../components/LessonDateDisplay/LessonDateDisplay";
 import LessonInfoDisplay from "../../components/LessonInfoDisplay/LessonInfoDisplay";
 import LessonNameDisplay from "../../components/LessonNameDisplay/LessonNameDisplay";
 
 const CustomLessonsDisplay = props => {
   const { firebase } = props;
-  const [adminLessons, setAdminLessons] = useState([]);
+  const [showLessons, setShowLessons] = useState([]);
+  const [allLessons, setAllLessons] = useState([]);
+  const [filterType, setFilterType] = useState();
+  const [filterGroup, setFilterGroup] = useState();
   const [displayLesson, setDisplayLesson] = useState(null);
   const [displayLessonTemplate, setDisplayTemplate] = useState(null);
-  const [displayLessonStudents, setDisplayStudents] = useState([]);
   const [displayLessonInfo, setDisplayLessonInfo] = useState(false);
   const [createLessonRedirect, setCreateLessonRedirect] = useState(false);
+  const [nameMap, setNameMap] = useState({});
   const editLessonRedirect = props?.location.state?.redirect;
 
-  function orderAdminLessons(reverse) {
-    const sortedLessons = [...adminLessons].sort((a, b) => {
-      if (reverse) {
-        return b.dueDate["seconds"] - a.dueDate["seconds"];
-      } else {
-        return a.dueDate["seconds"] - b.dueDate["seconds"];
-      }
-    });
-    setAdminLessons(sortedLessons);
-  }
+  /**
+   *
+   * wait can teachers delete whole lessons that they've already made?????
+   */
 
   useEffect(() => {
     setTimeout(() => {
       // TODO: figure out better solution to resolve this
       // Get custom lessons made by admin
-      firebase.getAdminCustomLessons(ADMIN_ACCOUNT).then(lesson => {
-        setAdminLessons(lesson);
+      firebase.getAdminCustomLessons(ADMIN_ACCOUNT).then(lessons => {
+        setAllLessons(lessons);
       });
-    }, 50); // Timeout for firebase to update and display updated lessons properly
+    }, 60); // Timeout for firebase to update and display updated lessons properly
   }, [editLessonRedirect]); // Updates lessons when redirected to page from CreateAssignment
 
+  //taken from lam
+  async function deploymentNameMap(lesson) {
+    let deploymentAccountIds = new Set();
+    for (const dueDate in lesson.dueDates) {
+      for (const deploymentAccount of lesson.dueDates[dueDate]) {
+        deploymentAccountIds.add(deploymentAccount);
+      }
+    }
+
+    let deploymentNameMap = {};
+    Array.from(deploymentAccountIds).forEach(id => {
+      firebase
+        .getDeploymentAccountInformation(id)
+        .then(deploymentAccount => {
+          deploymentNameMap[id] = deploymentAccount.username;
+        })
+        .catch(error => console.error("Error getting custom lesson: ", error));
+    });
+
+    //Wait for deploymentNameMap to finish resolving
+    while (Object.keys(deploymentNameMap).length == 0) {
+      await new Promise(r => setTimeout(r, 500));
+    }
```

Comment:
lam fixed this jank code actually so it would be great if you took it from him again